keps/sig-windows/windows-runtimeclass-support.md

```diff
@@ -0,0 +1,555 @@
+
+### Test Plan
+
+
+#### E2E Testing with CRI-ContainerD and Kubernetes
+
+We already have E2E tests targeting the CRI-ContainerD & Kubernetes integration running on Windows - see the KEP for [Windows CRI-ContainerD] for more details. We will add a few `RuntimeClass` definitions to those E2E tests to confirm that Pods can run with `runtime_handler` specified, and without it specified.
+
+As ContainerD integration itself is an alpha feature, we'll be experimenting with these configurations and deciding on a set of recommended defaults before graduating to beta. These defaults will be tested at beta with new test cases.
+
+The existing tests relying on `dockershim` will be run side by side until it is deprecated. This timeline hasn't been specified yet, but it will be announced before CRI-ContainerD for Windows is declared stable in 1.19 or later and follow a normal deprecation cycle.
+
+#### Unit testing with CRITest
+
+Unit tests will be added in CRITest which is in the [Cri-Tools] repo. Tests are already running on Windows - see [testgrid](https://k8s-testgrid.appspot.com/sig-node-containerd#cri-validation-windows).
+
+
+### Graduation Criteria
+
+#### Alpha - Kubernetes 1.17
+
+This timeline should follow that of [Windows CRI-ContainerD].
+
+In addition to what's included there, for alpha: 
+
+- Unit tests will be added to critest for `runtime_handler`
+- E2E tests for basic coverage of:
+  - new label set by kubelet
+
+#### Alpha -> Beta Graduation
+
+This timeline should follow that of [Windows CRI-ContainerD].
+
+In addition to what's included there, for beta: 
+
+- Unit tests will be added to critest
+- E2E tests for basic coverage of:
+  - `runtime_handler` updates
+
+#### Beta -> GA Graduation
+
+- Define & add E2E testing for recommended default RuntimeClass configurations for Windows supported with CRI-ContainerD
+
+
+##### Removing a deprecated flag
+
+- Announce deprecation and support policy of the existing flag
+- Two versions passed since introducing the functionality which deprecates the flag (to address version skew)
+- Address feedback on usage/changed behavior, provided on GitHub issues
+- Deprecate the flag
+
+**For non-optional features moving to GA, the graduation criteria must include [conformance tests].**
+
+[conformance tests]: https://github.com/kubernetes/community/blob/master/contributors/devel/conformance-tests.md
+
+### Upgrade / Downgrade Strategy
+
+The new label `node.kubernetes.io/osversion` can be set or removed if needed without impacting other components as described in [Risks and Mitigations](#risks-and-mitigations)
+
+Users can only opt-in to use the new `runtime_handler` field after setting up and configuring ContainerD. On existing clusters without ContainerD set up, they must use `docker` as the `runtimeHandler` in the `RuntimeClass` today. Therefore they must update to a supported version of ContainerD as a prerequisite which is covered in the scope of another KEP - [Windows CRI-ContainerD].
+
+
```

Comment:
Add section - future considerations based on SIG-Node feedback.

1. RuntimeClass parameters map - instead of encoding things within a RuntimeClass name, should we add some fields to RuntimeClass itself? If so, that could simplify RuntimeClass usage and also build some consistency across runtimes
2. PodOverhead - this also had a use case that may require setting some hypervisor parameters

Reply:
done